pennylane/ops/op_math/pow.py

```diff
@@ -216,6 +216,21 @@ def __init__(self, base=None, z=1, do_queue=True, id=None):
 
         super().__init__(base, do_queue=do_queue, id=id)
 
+        if isinstance(self.z, int) and self.z > 0:
```

Comment:
We could? I decide to go this route because I thought we could come back and move more logic to this layer of abstraction for the composite class. I don't see any similar logic we can pull out for individual ops so I think we can keep it as is for now.